### PR TITLE
Fix dangling pointers

### PR DESCRIPTION
**Pull request by xsellier**
Fixes https://github.com/godotengine/godot/issues/41682

Tested on:
- [x] Linux
- [x] Windows
- [x] OSX

Tests list:
- Export City Game Studio
- Run City Game Studio on debug/release
- Check for performance regressions
- Let it run for a while (~1h)

### CODE REVIEW

File: core/object_id.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/*************************************************************************/
+
+#ifndef OBJECTID_H
+#define OBJECTID_H
+
+#include "core/int_types.h"
+
+typedef uint32_t ObjectID;
+
+#endif
```
Comment (SecretPanda420) [inline core/object_id.h:38]: [New line missing.](https://stackoverflow.com/a/729795)

**Review by SecretPanda420 — CHANGES_REQUESTED** [review]
EOL at EOF missing.

File: core/object_id.h (hunk 1 above)
Comment (xsellier) [reply core/object_id.h]: Thanks, didn't notice. Just amened the commit

**Comment by xsellier** [comment]
Closed because the v2.1 won't get a fix that changes its core files.

**Pull request closed without merging**